nixos/modules/config/xdg/portal.nix

```diff
@@ -54,13 +54,14 @@ with lib;
     in mkIf cfg.enable {
 
       assertions = [
-        { assertion = (cfg.gtkUsePortal -> cfg.extraPortals != []);
-          message = "Setting xdg.portal.gtkUsePortal to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
+        { assertion =  cfg.extraPortals != [];
+          message = "Setting xdg.portal.enable to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
         }
       ];
 
       services.dbus.packages  = packages;
       systemd.packages = packages;
+      environment.systemPackages= packages;
```

Comment:
What is the point of this? The D-Bus and systemd services should already be handled by the options above?

Reply:
Also, looks like the space preceding the equal sign got teleported after the one next to the `assertion` :stuck_out_tongue_winking_eye:

Reply:
Hmm, there appears to be a [dummy desktop file](https://github.com/flatpak/xdg-desktop-portal-gtk/commit/3fd9e6715a98a71c09eea0f6eb00888a7ecd1970) that would warrant this. Could you please add a comment about that?

But I do not see how would this fix anything other than the issue described in the commit message introducing the desktop file.

Reply:
No, those are still necessary (at least one of them is but I prefer to be explicit). I would just add a comment explaining why `systemPackages` is needed since it is a catch-all.

Reply:
The problem is that I don't know _why_ this is fixing anything.

Reply:
Could it be related to the desktop file mentioned in the xdg-desktop-portal-gtk PR linked in the comment?

Reply:
Maybe, now that I've checked it there is a share/applications/org.freedesktop.impl.portal.desktop.kde.desktop

Reply:
Considering how much wayland relies on desktop files for stuff in general that would make sense. And without this line that file does indeed not exist.

Reply:
I do not think the desktop file does anything other than the mentioned window to icon association. If the desktop file was responsible, installing the following should also fix it:

```nix
(pkgs.runCommand "xdg-desktop-portal-desktops" {} ''
  mkdir -p "$out/share/applications/"
  cp "${pkgs.xdg-desktop-portal-gtk}/share/applications"/* "$out/share/applications/"
  cp "${pkgs.xdg-desktop-portal-kde}/share/applications"/* "$out/share/applications/"
'')
```

Reply:
You mean like this?

```nix
environment.systemPackages = [(pkgs.runCommand "xdg-desktop-portal-desktops" {} ''
  mkdir -p "$out/share/applications/"
  cp "${pkgs.xdg-desktop-portal-gtk}/share/applications"/* "$out/share/applications/"
  cp "${pkgs.xdg-desktop-portal-kde}/share/applications"/* "$out/share/applications/"
'')];
```

That way it does, in fact, work. The desktop file contains the full path to the portal executable:
```
Exec=/nix/store/i6rxcrs3bzy91hvjrs76g7d5jdqlg2f3-xdg-desktop-portal-kde-5.23.4/libexec/xdg-desktop-portal-kde
```
FWIW, this location is NOT in `PATH`.

Reply:
I'm not very familiar with wayland/kwin workings but even if other DEs don't require desktop files for screen sharing to work, that remains my primary suspect. Could be #JustKwinThings, but it would definitely be nice to get some confirmation from upstream.

Reply:
Now that I've looked at it there is also the `xdg-desktop-portal/portals/kde.portal` and the `xdg-desktop-portal-kde/qml/*`. This without talking about the `locale` directory …

Reply:
`xdg-desktop-portal/portals/kde.portal` is found in `$XDG_DESKTOP_PORTAL_DIR` so that should be accounted for.